MiniApp/Classes/API/ListingAPI.swift

```diff
@@ -0,0 +1,27 @@
+internal class ListingApi {
+    let environment: Environment
+
+    init(environment: Environment) {
+        self.environment = environment
+    }
+
+    func urlRequest() -> URLRequest? {
+        guard let url = getListingURL() else {
+            return nil
+        }
+        return listingURLRequest(url: url)
+    }
+
+    func getListingURL() -> URL? {
```

Comment:
I think `getListingURL` and `listingURLRequest` can become file-private, what do you think?

Reply:
Done